Mathlib/NumberTheory/LSeries/Basic.lean

```diff
@@ -12,140 +12,216 @@ import Mathlib.Analysis.NormedSpace.FiniteDimension
 -/
 
 
 open scoped BigOperators
 
-namespace ArithmeticFunction
+open Complex
+
+/-!
+### The terms of an L-series
+
+We define the `n`th term evaluated at a complex number `s` of the L-series associated
+to a sequence `f : ℕ → ℂ`, `LSeries.term f s n`, and provide some basic API.
 
-open Complex Nat
+We set `LSeries.term f s 0 = 0`, and for positive `n`, `LSeries.term f s n = f n / n ^ s`.
+-/
+
+namespace LSeries
 
-/-- The L-series of an `ArithmeticFunction`. -/
+/-- The `n`th term of the L-series of `f` evaluated at `s`. We set it to zero when `n = 0`. -/
 noncomputable
-def LSeries (f : ArithmeticFunction ℂ) (z : ℂ) : ℂ :=
-  ∑' n, f n / n ^ z
-#align nat.arithmetic_function.l_series ArithmeticFunction.LSeries
+def term (f : ℕ → ℂ) (s : ℂ) (n : ℕ) : ℂ :=
+  if n = 0 then 0 else f n / n ^ s
+
+lemma term_def (f : ℕ → ℂ) (s : ℂ) (n : ℕ) :
+    term f s n = if n = 0 then 0 else f n / n ^ s :=
+  rfl
+
+@[simp]
+lemma term_zero (f : ℕ → ℂ) (s : ℂ) : term f s 0 = 0 := rfl
+
+-- We put `hn` first for convnience, so that we can write `rw [LSeries.term_of_ne_zero hn]` etc.
+@[simp]
+lemma term_of_ne_zero {n : ℕ} (hn : n ≠ 0) (f : ℕ → ℂ) (s : ℂ) :
+    term f s n = f n / n ^ s :=
+  if_neg hn
+
+lemma term_congr {f g : ℕ → ℂ} (h : ∀ n ≠ 0, f n = g n) (s : ℂ) (n : ℕ) :
+    term f s n = term g s n := by
+  rcases eq_or_ne n 0 with hn | hn <;> simp [hn, h]
+
+lemma norm_term_eq (f : ℕ → ℂ) (s : ℂ) (n : ℕ) :
+    ‖term f s n‖ = if n = 0 then 0 else ‖f n‖ / n ^ s.re := by
+  rcases eq_or_ne n 0 with rfl | hn
+  · simp only [term_zero, norm_zero, ↓reduceIte]
+  · rw [if_neg hn, term_of_ne_zero hn, norm_div, norm_natCast_cpow_of_pos <| Nat.pos_of_ne_zero hn]
+
+lemma norm_term_le {f g : ℕ → ℂ} (s : ℂ) {n : ℕ} (h : ‖f n‖ ≤ ‖g n‖) :
+    ‖term f s n‖ ≤ ‖term g s n‖ := by
+  simp only [norm_term_eq]
+  split
+  · rfl
+  · gcongr
+
+lemma norm_term_le_of_re_le_re (f : ℕ → ℂ) {s s' : ℂ} (h : s.re ≤ s'.re) (n : ℕ) :
+    ‖term f s' n‖ ≤ ‖term f s n‖ := by
+  simp only [norm_term_eq]
+  split
+  next => rfl
+  next hn => gcongr; exact Nat.one_le_cast.mpr <| Nat.one_le_iff_ne_zero.mpr hn
+
+end LSeries
+
+/-!
+### Definition of the L-series and related statements
+
+We define `LSeries f s` of `f : ℕ → ℂ` as the sum over `LSeries.term f s`.
+We also provide predicates `LSeriesSummable f s` stating that `LSeries f s` is summable
+and `LSeriesHasSum f s a` stating that the L-series of `f` is summable at `s` and converges
+to `a : ℂ`.
+-/
 
-/-- `f.LSeriesSummable z` indicates that the L-series of `f` converges at `z`. -/
-def LSeriesSummable (f : ArithmeticFunction ℂ) (z : ℂ) : Prop :=
-  Summable fun n => f n / n ^ z
-#align nat.arithmetic_function.l_series_summable ArithmeticFunction.LSeriesSummable
+open LSeries
 
-theorem LSeries_eq_zero_of_not_LSeriesSummable (f : ArithmeticFunction ℂ) (z : ℂ) :
-    ¬f.LSeriesSummable z → f.LSeries z = 0 :=
+/-- The value of the L-series of the sequence `f` at the point `s`
```

Comment:
if it converges absolutely, and 0 otherwise.

Reply:
OK.